File: strykerjs-node-test-poc/src/stringUtils.ts
export function reverse(s: string): string {
  return s.split('').reverse().join('');
}

export function isPalindrome(s: string): boolean {
  const cleaned = s.toLowerCase().replace(/[^a-z0-9]/g, '');
  return cleaned === reverse(cleaned);
}

export function capitalize(s: string): string {
  if (s.length === 0) return s;
  return s[0].toUpperCase() + s.slice(1);
}

export function countVowels(s: string): number {
  const matches = s.toLowerCase().match(/[aeiou]/g);
  return matches ? matches.length : 0;
}

File: stryker4s-poc/src/main/scala/poc/StringUtils.scala
package poc

object StringUtils {

  def reverse(s: String): String = s.reverse

  def isPalindrome(s: String): Boolean = {
    val cleaned = s.toLowerCase.filter(_.isLetterOrDigit)
    cleaned == cleaned.reverse
  }

  def capitalize(s: String): String =
    if (s.isEmpty) s else s.head.toUpper + s.tail

  def countVowels(s: String): Int =
    s.toLowerCase.count(c => "aeiou".contains(c))
}
